docs/389ds/design/configurable_pbkdf2_iterations.md

```diff
@@ -0,0 +1,107 @@
+---
+title: "Configurable Number of Iterations for PBKDF2 Password Storage Schemes"
+---
+
+# Configurable Number of Iterations for PBKDF2 Password Storage Schemes
+
+## Overview
+This design document describes the implementation of configurable number of iteration (rounds) for PBKDF2 password hashing schemes in 389 Directory Server. The feature adds the ability to customize the computational cost of password hashing through a new attribute `nsslapd-pwdPBKDF2NumIterations` and associated configuration mechanisms.
+
+## Background
+PBKDF2 (Password-Based Key Derivation Function 2) is a key stretching algorithm that makes password hashing more resistant to brute-force attacks by applying a cryptographic hash function repeatedly. The number of iterations directly affects the computational cost and security level of the password hashing process. As computing power increases over time, the ability to adjust iteration counts becomes crucial for maintaining security against evolving threats.
+
+## Technical Design
+
+### Schema Changes
+1. **New Attribute**:
+   ```ldif
+   attributeTypes: ( 2.16.840.1.113730.3.1.2400
+       NAME 'nsslapd-pwdPBKDF2NumIterations'
+       DESC '389 Directory Server defined attribute type'
+       SYNTAX 1.3.6.1.4.1.1466.115.121.1.27
+       SINGLE-VALUE )
+   ```
+
+2. **New ObjectClass**:
+   ```ldif
+   objectClasses: ( 2.16.840.1.113730.3.2.340
+       NAME 'pwdPBKDF2PluginConfig'
+       DESC 'PBKDF2 Password Storage Plugin configuration'
+       SUP top
+       MAY ( nsslapd-pwdPBKDF2NumIterations ) )
+   ```
```

Comment:
You may indicate which entries will get this new objectclass (like 'cn=PBKDF2*,cn=Password Storage Schemes,cn=plugins,cn=config').